openedx/core/djangoapps/video_pipeline/tests/test_api.py

```diff
@@ -0,0 +1,98 @@
+"""
+Tests for Video Pipeline api utils.
+"""
+import ddt
+from mock import Mock, patch
+
+from django.test.testcases import TestCase
+from slumber.exceptions import HttpClientError
+
+from student.tests.factories import UserFactory
+
+from openedx.core.djangoapps.video_pipeline.api import update_3rd_party_transcription_service_credentials
+from openedx.core.djangoapps.video_pipeline.tests.mixins import VideoPipelineIntegrationMixin
+
+
+@ddt.ddt
+class TestAPIUtils(VideoPipelineIntegrationMixin, TestCase):
+    """
+    Tests for API Utils.
+    """
+    def setUp(self):
+        self.pipeline_integration = self.create_video_pipeline_integration()
+        self.user = UserFactory(username=self.pipeline_integration.service_username)
+
+    def test_update_transcription_service_credentials_with_integration_disabled(self):
+        """
+        Test updating the credentials when service integration is disabled.
+        """
+        self.pipeline_integration.enabled = False
+        self.pipeline_integration.save()
+        is_updated = update_3rd_party_transcription_service_credentials()
+        self.assertFalse(is_updated)
+
+    def test_update_transcription_service_credentials_with_unknown_user(self):
+        """
+        Test updating the credentials when expected service user is not registered.
+        """
+        self.pipeline_integration.service_username = 'non_existent_user'
+        self.pipeline_integration.save()
+        is_updated = update_3rd_party_transcription_service_credentials()
+        self.assertFalse(is_updated)
+
+    @ddt.data(
+        (
+            {
+                'username': 'Jason_cielo_24',
+                'api_key': '12345678',
+            },
+            Mock(),
```

Comment:
Why are we creating Mock() here? I see you dont do anything extra to mocks in the data, adding it in the method body would be more better

Reply:
In one of these cases, there is side effect, that's why.

Reply:
Oh, I've extracted that out as I needed to assert exception message as well. I will surely remove these Mocks.